xformers/ops/fmha/triton_splitk.py

```diff
@@ -1169,21 +1169,34 @@ def not_supported_reasons(cls, d: Inputs) -> List[str]:
     def get_split_k(cls, B: int, G: int, H: int, Mk: int) -> int:
         """Heuristic for the number of splits"""
         bh = max(B * H, 1)  # NOTE: Handle B*h=0 case
-        split_k = max(Mk, 1024) // bh
+        split_k = max(Mk + bh - 1, 1024) // bh
         if torch.version.hip:
             max_chunk_size = 64
-            split_k_stop_val = min(Mk / max_chunk_size, 1024 / (B * G * H))
+            split_k_stop_val = 1024 / (B * G * H)
+            while split_k > 0 and Mk / (split_k - 1) < max_chunk_size:
```

Comment:
Should it be while split_k > 1 to prevent division by 0?

Reply:
Thanks, yes, good catch, should be `split_k > 1`.